presto-tablestore/src/main/java/com/facebook/presto/tablestore/ReflectionUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tablestore;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.UndeclaredThrowableException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public abstract class ReflectionUtils
+{
+    public static Field findField(Class<?> clazz, String name)
+    {
+        return findField(clazz, name, null);
+    }
+
+    public static Field findField(Class<?> clazz, String name, Class<?> type)
+    {
+        checkArgument(clazz != null, "Class must not be null");
+        checkArgument(name != null || type != null, "Either name or type of the field must be specified");
+        Class<?> searchType = clazz;
+        while (!Object.class.equals(searchType) && searchType != null) {
+            Field[] fields = searchType.getDeclaredFields();
+            for (Field field : fields) {
+                if ((name == null || name.equals(field.getName())) && (type == null || type.equals(field.getType()))) {
+                    return field;
+                }
+            }
+            searchType = searchType.getSuperclass();
+        }
+        return null;
+    }
+
+    public static Object getField(Field field, Object target)
+    {
+        try {
+            return field.get(target);
+        }
+        catch (IllegalAccessException ex) {
+            handleReflectionException(ex);
```

Comment:
why do we need this handleReflectionException method? I might prefer to use multiple catch clauses here which looks more straight forward.